components/bl-transferList-component/src/list/index.js

```diff
@@ -0,0 +1,49 @@
+import { ListHeader } from './list-header';
+import { ListItem } from './list-item';
+
+export const List = props => {
+  const { title, iconColor, enableSelectAll, items, selected, allSelected, setAllSelected } = props;
+
+  const handleToggleItem = ({ target: { value, checked } }) => {
+    const item = items.find(item => item.value === value);
+
+    if (checked) {
+      setAllSelected(prevState => [...prevState, item]);
+    } else {
+      const newAllSelected = allSelected.filter(({ objectId }) => item.objectId !== objectId);
+
+      setAllSelected(newAllSelected);
+    }
+  };
+
+  return (
+    <div className="list">
+      { enableSelectAll &&
+        <ListHeader
+          items={ items }
+          title={ title }
+          selected={ selected }
+          iconColor={ iconColor }
+          setAllSelected={ setAllSelected }
+        />
+      }
+      <div className="list__items">
+        { items.map(item => {
+          const isChecked = allSelected.some(({ objectId }) => item.objectId === objectId);
+          
+          return (
+            <ListItem
+              key={ item.objectId }
+              id={ item.objectId }
```

Comment:
what if the list item doesn't contain an `objectId`?

Reply:
In the case that data without an objectId is received, the validation function will add an objectId.